session/session.go

```diff
@@ -2374,7 +2374,7 @@ func (s *session) ExecutePreparedStmt(ctx context.Context, stmtID uint32, args [
 		if err != nil {
 			return nil, err
 		}
-	} else if preparedStmt.ForUpdateRead {
+	} else if s.sessionVars.IsIsolation(ast.ReadCommitted) || preparedStmt.ForUpdateRead {
 		is = domain.GetDomain(s).InfoSchema()
```

Comment:
Here we should use `temptable.AttachLocalTemporaryTableInfoSchema` to wrap the infoschema because temporary tables may exist in session